pilot/pkg/proxy/envoy/v2/rds.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"fmt"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/gogo/protobuf/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+func (s *DiscoveryServer) pushRoute(con *XdsConnection) error {
+	rawRoutes, err := s.generateRawRoutes(con)
+	if err != nil {
+		return err
+	}
+	for _, r := range rawRoutes {
+		con.RouteConfigs[r.Name] = r
+		resp, _ := model.ToJSONWithIndent(r, " ")
+		adsLog.Debugf("RDS: Adding route %s for node %s", resp, con.modelNode)
+	}
+
+	response := routeDiscoveryResponse(rawRoutes, *con.modelNode)
+	err = con.send(response)
+	if err != nil {
+		adsLog.Warnf("ADS: RDS: Send failure for %s, closing grpc %v", con.modelNode, err)
+		pushes.With(prometheus.Labels{"type": "rds_senderr"}).Add(1)
+		return err
+	}
+	pushes.With(prometheus.Labels{"type": "rds"}).Add(1)
+
+	adsLog.Infof("ADS: RDS: PUSH for node: %s addr:%s routes:%d", con.modelNode, con.PeerAddr, len(rawRoutes))
+	return nil
+}
+
+func (s *DiscoveryServer) generateRawRoutes(con *XdsConnection) ([]*xdsapi.RouteConfiguration, error) {
+	rc := make([]*xdsapi.RouteConfiguration, 0)
+	// TODO: Follow this logic for other xDS resources as well
+	// And cache/retrieve this info on-demand, not for every request from every proxy
```

Comment:
Can we take this out? If someone needs this commented out code we can bring it back since its already in source control!

Reply:
These aren't my comments I just split out what we had into a separate file. We could ask whoever wrote them in the first place though I guess?

Reply:
it was @rshriram :D. @rshriram can we please delete these and bring them back when needed? Either that, or lets put a bug in for caching so we can track this via github.

Reply:
sure go for it.